src/components/Tree/TreeItem.tsx

```diff
@@ -44,50 +87,122 @@ export const TreeItem = ({
-        event.stopPropagation();
-        setExpanded((previousExpanded) => !previousExpanded);
-    };
+    const handleExpandKeyDown = useCallback(
+        (event: KeyboardEvent) => {
+            if (event.key === 'Space' || event.key === 'Enter') {
+                event.stopPropagation();
+                onExpand(id, !treeState.expandedIds.has(id));
+            }
+        },
+        [id, onExpand, treeState.expandedIds],
+    );
 
-    const childrenArray = React.Children.toArray(children);
+    const handleKeyDown = useCallback(
+        (event: KeyboardEvent) => {
+            switch (event.code) {
+                case 'Enter':
+                    event.preventDefault();
+                    onSelect(id);
+                    break;
 
-    let enhancedChildren: ReactNode = childrenArray;
+                case 'Space':
+                    event.preventDefault();
+                    if (treeItemState?.childrenIds) {
+                        onExpand(id, !treeState.expandedIds.has(id));
+                    } else {
+                        onSelect(id);
+                    }
+                    break;
 
-    if (draggable) {
-        enhancedChildren = draggableEnhancedChildren;
-    }
+                case 'ArrowRight':
+                    event.preventDefault();
+                    if (treeItemState?.childrenIds && !treeState.expandedIds.has(id)) {
+                        onExpand(id, true);
+                    } else if (treeItemState?.childrenIds?.[0]) {
+                        const firstChildId = treeItemState.childrenIds[0];
+                        const firstChildState = treeState.items.get(firstChildId);
+                        firstChildState?.domElement?.focus();
+                    }
+                    break;
 
-    const caretComponent =
-        childrenArray.length > 0 ? (
-            <div
-                className={merge([
-                    'tw-transition-transform tw-w-0 tw-h-0 tw-text-black-100 tw-text-opacity-40 tw-font-normal tw-border-t-4 tw-border-t-transparent tw-border-b-4 tw-border-b-transparent tw-border-l-4 tw-border-l-x-strong',
-                    expanded ? 'tw-rotate-90' : '',
-                    selectedIds.includes(id) && 'tw-text-box-selected-strong-inverse',
-                ])}
-            />
-        ) : (
-            <></>
-        );
+                case 'ArrowLeft':
+                    event.preventDefault();
+                    if (treeItemState?.childrenIds && treeState.expandedIds.has(id)) {
+                        onExpand(id, false);
+                    } else if (treeItemState?.parentId) {
+                        const parentId = treeItemState.parentId;
+                        const parentState = treeState.items.get(parentId);
+                        parentState?.domElement?.focus();
+                    }
+                    break;
+
+                case 'ArrowUp':
+                    event.preventDefault();
```

Comment:
event.preventDefault() can be put on top

Reply:
It prevents tabbing